javalib/src/main/scala/java/util/logging/Formatter.scala

```diff
@@ -0,0 +1,22 @@
+package java.util.logging
+
+abstract class Formatter protected () {
+
+  def format(record: LogRecord): String
+
+  def getHead(h: Handler): String = ""
+
+  def getTail(h: Handler): String = ""
+
+  def formatMessage(record: LogRecord): String =
+    // The Java spec uses java.text formatting not available in Scala.js
+    if (Option(record.getParameters).exists(_.nonEmpty)) {
+      // Simple text replacement
+      val msg = record.getMessage
+      record.getParameters.zipWithIndex.foldRight(msg) { case ((x, i), t) =>
+        t.replace(s"{$i}", x.toString)
```

Comment:
This transformation will create a linear amount of `Strings` (in respect to number of parameters). Consider using a `StringBuilder` and a single pass through `msg`.